app/src/main/java/com/dgaf/happyhour/Controller/DrawerFragment.java

```diff
@@ -328,27 +395,20 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean enabled) {
                     queryParameters.getDayOfWeekMask().unselectSunday();
                     break;
                 case R.id.today:
-                    //only unselect today if another day is selected
-                    if(isADaySelected()) {
-                        compoundButton.setBackgroundResource(R.drawable.ic_today);
-                        queryParameters.getDayOfWeekMask().unselectToday();
-                    }
+                    compoundButton.setBackgroundResource(R.drawable.ic_today);
+                    queryParameters.getDayOfWeekMask().unselectToday();
                     break;
             }
         }
     }
 
     private void setTitle(String s) {
-    }
 
-    private void disableTodayToggle() {
-        today.setChecked(false);
-        today.setBackgroundResource(R.drawable.ic_today);
-        queryParameters.getDayOfWeekMask().unselectToday();
     }
 
     @Override
-    public void notifyEmpty() {}
+    public void notifyEmpty() {
```

Comment:
We have to. Its part of the interface we are implementing to get the toast feedback.